src/layout/Grid.tsx

```diff
@@ -43,57 +47,70 @@ const matchLargeDesktop = window.matchMedia(LARGE_DESKTOP_MEDIA_QUERY);
 // Currently this grid simply changes the number of columns that are used based on a media query
 // The thinking for this is that we want to be able to change the positioning of child elements based on
 // the "breakpoints" of the changing grid
-export const Grid = ({ childrenForMobileDisplay, childrenForTabletDisplay, childrenForSmallDesktopDisplay, childrenForLargeDesktopDisplay }: GridProps) => {
-
-    // This feels potentially expensive being here :/
-    // Maybe I should be setting up the matchMedia query just once outside the function and then calling it
-    const getMedia: GetMedia = () => {
-        if (matchLargeDesktop.matches) {
-            return 'LARGE_DESKTOP';
-        }
-        if (matchSmallDesktop.matches) {
-            return 'SMALL_DESKTOP';
-        }
-        if (matchTablet.matches) {
-            return 'TABLET';
-        }
-        if (matchMobile.matches) {
-            return 'MOBILE';
-        }
-        return 'UNSUPPORTED';
+export const Grid = ({
+  childrenForMobileDisplay,
+  childrenForTabletDisplay,
+  childrenForSmallDesktopDisplay,
+  childrenForLargeDesktopDisplay,
+}: GridProps) => {
+  // This feels potentially expensive being here :/
+  // Maybe I should be setting up the matchMedia query just once outside the function and then calling it
+  const getMedia: GetMedia = () => {
+    if (matchLargeDesktop.matches) {
+      return "LARGE_DESKTOP";
     }
-
-    // Set the initial state for the media type
-    const [media, setMedia] = useState(
-        getMedia()
-    )
-
-    useEffect(() => {
-        // If this is running on every render then it suggests that we're just repeatedly creating listeners, which would be bad :/ 
-        // Need to check and then fix this if that's the case...
-        matchMobile.addEventListener('change', () => setMedia(getMedia()));
-        matchTablet.addEventListener('change', () => setMedia(getMedia()));
-        matchSmallDesktop.addEventListener('change', () => setMedia(getMedia()));
-        matchLargeDesktop.addEventListener('change', () => setMedia(getMedia()));
-    }, []); // <- I think that this empty array solves the efficiency problem by only running on mount and unmount, but I need to check
-
-
-    const getClassName = (media: Media) => {
-        return `grid ${media.toLowerCase()}`;
+    if (matchSmallDesktop.matches) {
+      return "SMALL_DESKTOP";
+    }
+    if (matchTablet.matches) {
+      return "TABLET";
+    }
+    if (matchMobile.matches) {
+      return "MOBILE";
+    }
+    return "UNSUPPORTED";
+  };
+
+  // Set the initial state for the media type
+  const [media, setMedia] = useState(getMedia());
+
+  useEffect(() => {
+    // If this is running on every render then it suggests that we're just repeatedly creating listeners, which would be bad :/
+    // Need to check and then fix this if that's the case...
+    matchMobile.addEventListener("change", () => setMedia(getMedia()));
```

Comment:
https://hello.atlassian.net/wiki/spaces/~akorzunov/blog/2021/10/22/1380136629/Match+Media+and+other+Shenanigans / https://github.com/thearnica/react-media-match